src/extension.ts

```diff
@@ -188,26 +197,10 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         await executeOrganizeImports(lsClient, serverId);
       }
     }),
-    registerCommand(`${serverId}.debugInformation`, async () => {
-      let configuration = getConfiguration(serverId) as unknown as ISettings;
-      if (!lsClient || !configuration.nativeServer) {
-        return;
-      }
-
-      const editor = vscode.window.activeTextEditor;
-      const params = {
-        command: `${serverId}.printDebugInformation`,
-        arguments: [
-          {
-            textDocument: editor ? { uri: editor.document.uri.toString() } : null,
-          },
-        ],
-      };
-
-      await lsClient.sendRequest(ExecuteCommandRequest.type, params).then(undefined, async () => {
-        vscode.window.showErrorMessage("Failed to print debug information.");
-      });
-    }),
+    registerCommand(
+      `${serverId}.debugInformation`,
+      createDebugInformationProvider(getClient, serverId, context),
+    ),
```

Comment:
Would it be possible to avoid creating the command if it is the native server?

Reply:
I don't think so. Looking at https://code.visualstudio.com/api/references/contribution-points#contributes.commands, we need to add it to the UI via `package.json` which makes this not possible (https://github.com/astral-sh/ruff-vscode/blob/b4e06981ac3148b3f08c074a759c68f7309cc536/package.json#L434-L438).